Round cleaning and freezetime recognition improvements.

JanEricNitschke opened the pull request:
Closes https://github.com/pnxenopoulos/awpy/issues/284

This PR includes a reduction of `remove_warmups` to it simplest form due to it seeming too rigid and functionally unclear.
It now only removes rounds tagged as `isWarmup`.

This was the main reason the demo in the issue had all of its rounds removed in cleaning.

---

I also adjusted `remove_bad_scoring` slightly so that it now requires rounds that are kept because they have a winning score to also have to have a higher score than the previous round.

In the cevo demo there were multiple rounds before the "last" that had the final game score.

16-5 (actual last round)
16-5 (some left over non gameplay round)
0-0 (reset round before the server closed)

As this check was not present previously this caused the "bad" second to last round to be wrongly kept. This has been fixed now.

---

Lastly the PR includes a change to freezetime/roundstart handling.

The `FreezeTimeEndHandler` handler includes a check whether the game is already (still) in freezetime to deal with cases where the `RoundStart` event did not fire. It subsequently adds the currently still  kept alive round to the list of game rounds and starts the new one from the freeze time end instead of the ideal round start.

The addition of the gamestate check for the freeze time which was added in `https://github.com/pnxenopoulos/awpy/pull/278` to handle cases where the `FrezeTimeEnd` event was missing. 

This introduced an issue when the gamestate changed before the event triggered. Causing the `roundInFreezetime` flag to be set before the event triggered which caused the event to add a lot short dummy rounds into the list of rounds.

I have now added an additional flag which indicates if the `roundInFreezetime` flag was set from the gamestate checker. If that is the case then the freezetime event handler does not add its own round.

These two flags could potentially be refactored into a single enum though. Dont know if you feel that that is necessary.

pnxenopoulos commented:
Mostly looks great, but want to hold off on merging just to get some clarification on the following: do we have a demo where the behavior described happens (might already exist in the test suite)?

JanEricNitschke commented:
So the issue i was trying to fix was https://github.com/pnxenopoulos/awpy/issues/284 which also has a link to the demo which we could add to the testing set.

The cause of that issue was that the warmup cleaning was removing all rounds.

Debugging that i also noticed that a bunch of dummy rounds got added (and cleaned) due to the freezetime changes.
While they were being cleaned they caused the roundstart time to be cut off (because the freezetime fallback only starts the rounds at the freezetime end).

So we could add that demo to the testing set and do a check on round numbers uncleaned and cleaned to verify that both these changes here work properly.
